server/zoom/api.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package zoom
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	jwlAlgorithm      = "HS256"
+	zoomEmailMismatch = "We could not verify your Mattermost account in Zoom. Please ensure that your Mattermost email address %s matches your Zoom login email address."
+)
+
+// APIClient represents a Zoom API client
+type APIClient struct {
+	apiKey     string
+	apiSecret  string
+	httpClient *http.Client
+	baseURL    string
+}
+
+// NewAPIClient returns a new Zoom API client
+func NewAPIClient(zoomAPIURL, apiKey, apiSecret string) *APIClient {
+	return &APIClient{
+		apiKey:     apiKey,
+		apiSecret:  apiSecret,
+		httpClient: &http.Client{},
+		baseURL:    zoomAPIURL,
+	}
+}
+
+func (c *APIClient) CompleteCompliance(payload DeauthorizationPayload) error {
+	req := complianceRequest{
+		ClientID:                     payload.ClientID,
+		UserID:                       payload.UserID,
+		AccountID:                    payload.AccountID,
+		DeauthorizationEventReceived: payload,
+		ComplianceCompleted:          true,
+	}
+
+	var ret string
+	return c.request(http.MethodPost, "/oauth/data/compliance", req, &ret)
+}
+
+func (c *APIClient) GetMeeting(meetingID int) (*Meeting, error) {
+	var meeting Meeting
+	err := c.request(http.MethodGet, fmt.Sprintf("/meetings/%v", meetingID), "", &meeting)
+	return &meeting, err
+}
+
+func (c *APIClient) GetUser(userID string) (*User, *AuthError) {
```

Comment:
Finally here we have to pass the user e-mail. We could pass the whole user here (so we can check the e-mail) or we can infer the e-mail here from the id (load the user from the api). I think it is easier to pass the whole user, but I am not 100% sure.